@Test public void whilePredicateThrows(){
  List<Throwable> errors=TestHelper.trackPluginErrors();
  try {
    Observable.just(1).forEachWhile(new Predicate<Integer>(){
      @Override public boolean test(      Integer v) throws Exception {
        throw new TestException();
      }
    }
);
    TestHelper.assertError(errors,0,OnErrorNotImplementedException.class);
    Throwable c=errors.get(0).getCause();
    assertTrue("" + c,c instanceof TestException);
  }
  finally {
    RxJavaPlugins.reset();
  }
}
